src/main/java/pl/byczazagroda/trackexpensesappbackend/service/UserServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package pl.byczazagroda.trackexpensesappbackend.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import pl.byczazagroda.trackexpensesappbackend.dto.AuthRegisterDTO;
+import pl.byczazagroda.trackexpensesappbackend.exception.AppRuntimeException;
+import pl.byczazagroda.trackexpensesappbackend.model.User;
+import pl.byczazagroda.trackexpensesappbackend.model.UserStatus;
+import pl.byczazagroda.trackexpensesappbackend.repository.UserRepository;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+
+import javax.validation.Valid;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@RequiredArgsConstructor
+@Service
+@Validated
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+
+    @Override
+    public void registerUser(@Valid AuthRegisterDTO authRegisterDTO) {
+        if (userRepository.existsByEmail(authRegisterDTO.email())) {
```

Comment:
I think it's better to verify the e-mail address beforehand, and only then refer to the repository to check it.